byway/src/main/webapp/destinations.js

```diff
@@ -215,10 +217,14 @@ function getCurrentAddress(){
 * add event listener for submit button
 */
 window.onload = function(){
+  let nextButton = document.getElementById('next-button');
+  nextButton.addEventListener('click', () => {
+      nextButton.href = "/interests.html?tripKey=" + tripKey;
```

Comment:
Same note on using `URLSearchParams`. If it seems repetitive, you could make a method like

```js
/** Adds a tripId parameter to a URL string. Assumes the URL has no existing parameters. */
function withTripIdParam(baseUrl, tripId) {
  return baseUrl + "?" + new URLSearchParams({tripId}).toString();
}

// Or even more generally:
/** Adds params to a string URL. Assumes the URL has no existing params. */
function withParams(baseUrl, params) {
  return baseUrl + "?" + new URLSearchParams(params).toString();
}

// Then you can use:
withParams("/interests.html", {tripId});
```

Reply:
added